tests/test_socket_server.py

```diff
@@ -0,0 +1,180 @@
+        await asyncio.sleep(0.5)
+        self.assertFalse(self.srv.connected)
+
+    def check_temperature_reply(
+        self, reply: typing.List[typing.Union[str, float]]
+    ) -> None:
+        device_name = reply[0]
+        time = reply[1]
+        response_code = reply[2]
+        resp = reply[3:]
+
+        self.assertEqual(self.name, device_name)
+        self.assertGreater(time, 0)
+        if self.in_error_state:
+            self.assertEqual(common.ResponseCode.DEVICE_READ_ERROR, response_code)
+        else:
+            self.assertEqual(common.ResponseCode.OK, response_code)
+        self.assertEqual(len(resp), self.num_channels)
+        for i in range(0, self.num_channels):
+            self.assertLessEqual(common.MockTemperatureConfig.min, resp[i])
+            self.assertLessEqual(resp[i], common.MockTemperatureConfig.max)
+
+    async def check_server_test(self) -> None:
+        """Test a full command sequence of the SocketServer.
+
+        The sequence is
+            - configure
+            - start
+            - read telemetry
+            - stop
+            - disconnect
+            - exit
+        """
+        configuration = {
+            common.Key.DEVICES: [
+                {
+                    common.Key.NAME: self.name,
+                    common.Key.CHANNELS: self.num_channels,
+                    common.Key.DEVICE_TYPE: common.DeviceType.FTDI,
+                    common.Key.FTDI_ID: "ABC",
+                    common.Key.SENSOR_TYPE: common.SensorType.TEMPERATURE,
+                }
+            ]
+        }
+        await self.write(
+            command=common.Command.CONFIGURE,
+            parameters={common.Key.CONFIGURATION: configuration},
+        )
+        data = await self.read()
+        self.assertEqual(common.ResponseCode.OK, data[common.Key.RESPONSE])
+        await self.write(command=common.Command.START, parameters={})
+        data = await self.read()
+        self.assertEqual(common.ResponseCode.OK, data[common.Key.RESPONSE])
+
+        self.reply = await self.read()
+        reply_to_check = self.reply[common.Key.TELEMETRY]
+        self.check_temperature_reply(reply_to_check)
+
+        self.reply = await self.read()
+        reply_to_check = self.reply[common.Key.TELEMETRY]
+        self.check_temperature_reply(reply_to_check)
+
+        await self.write(command=common.Command.STOP, parameters={})
+        data = await self.read()
+        self.assertEqual(common.ResponseCode.OK, data[common.Key.RESPONSE])
+        await self.write(command=common.Command.DISCONNECT, parameters={})
+        await self.write(command=common.Command.EXIT, parameters={})
+
+    async def test_full_command_sequence(self) -> None:
+        """Test the SocketServer with a nominal configuration, i.e. no
+        disconnected channels and no truncated data.
+        """
+        self.num_channels = 4
+        self.in_error_state = False
+        await self.check_server_test()
```

Comment:
I suggest making the two instance variables above into arguments to check_server_test (for the reason noted in the comment above about self.num_channels).